pkg/http/http_test.go

```diff
@@ -325,25 +324,33 @@ func TestMaxRetries(t *testing.T) {
 	testCases := []struct {
 		client       Client
 		countedCalls int
+		method       string
+		responseCode int
+		errorText    string
 	}{
-		{client: Client{maxRetries: 0}, countedCalls: 1},
-		{client: Client{maxRetries: 2}, countedCalls: 3},
-		{client: Client{maxRetries: 3}, countedCalls: 4},
+		{client: Client{maxRetries: 0}, countedCalls: 1, method: http.MethodGet, responseCode: 500, errorText: "Internal Server Error"},
+		{client: Client{maxRetries: 2}, countedCalls: 3, method: http.MethodGet, responseCode: 500, errorText: "Internal Server Error"},
+		{client: Client{maxRetries: 3}, countedCalls: 4, method: http.MethodPost, responseCode: 503, errorText: "Service Unavailable"},
+		{client: Client{maxRetries: 3}, countedCalls: 4, method: http.MethodPut, responseCode: 506, errorText: "Variant Also Negotiates"},
+		{client: Client{maxRetries: 3}, countedCalls: 4, method: http.MethodHead, responseCode: 502, errorText: "Bad Gateway"},
+		{client: Client{maxRetries: 3}, countedCalls: 1, method: http.MethodHead, responseCode: 404, errorText: "Not Found"},
+		{client: Client{maxRetries: 3}, countedCalls: 1, method: http.MethodHead, responseCode: 401, errorText: "Authentication Error"},
```

Comment:
[curiosity] not sure if I understand the details here right: but does it make sense to retry after a Authentication Error? That might lead to a (temporary) locked account.

Reply:
@marcusholl the retry is a general setting and the test ensures that no retry is performed and therefore `countedCalls` is just one instead of 4 ;-)